tests/foreman/ui/test_repository.py

```diff
@@ -942,10 +942,10 @@ def test_positive_recommended_repos(session, module_entitlement_manifest_org):
         cap_tool_repos = [
             repo['name']
             for repo in rrepos_on
-            if 'Tools' in repo['name'] or 'Capsule' in repo['name']
+            if 'Utils' in repo['name'] or 'Capsule' in repo['name']
         ]
         cap_tools_repos = [repo for repo in cap_tool_repos if repo.split()[4] != sat_version]
```

Comment:
I think instead of checking this for != sat_version, we should verify if n-1 repos are being suggested for capsule and utils repos

Reply:
these repos should always match the sat version. so if any do not, then we throw the error message

Reply:
I tested locally, as in updated code for variable `cap_utils_main_repos`, getting sat_version `n-1` value, as you suggested @Gauravtalreja1 
We just need to assert currect sat version and sat version found in repo names (in `cap_utils_main_repos` list) are not same.

Reply:
yes, I think with n-1 it will be more explicite and efficient assertion here, which would all repositories equal to n-1 versions instead of checking not equal to current sat versions

Reply:
@ColeHiggins2 ^

Reply:
@Gauravtalreja1  my observation
On one of older snap version of `6.15 snap 3.0` I had seen 6.14 repos of `capsule/maintain/Utils` but on latest snap verson of `6.15 snap 9.0` I couldn't see.

Reply:
I believe that on Satellite of version N only the N `capsule/maintain/utils` repos should be recommended. In other words, N-1, N-2, and all older repos should be missing.
@Gauravtalreja1, can you describe why you propose to check N-1 only?

Reply:
> @Gauravtalreja1 my observation On one of older snap version of `6.15 snap 3.0` I had seen 6.14 repos of `capsule/maintain/Utils` but on latest snap verson of `6.15 snap 9.0` I couldn't see.

This is expected behavior. at some point between snap 3 and 9, 6.14 repos were removed from recommended.